src/services/MosaicService.ts

```diff
@@ -68,14 +71,20 @@ export class MosaicService {
     'mosaicConfiguration')
 
   /**
-   * Store that caches the information around the network currency. The network currency is
-   * currently calculated from the block 1 transactions.
-   *
-   * In the near future, rest will return the information without loading block 1.
+   * Store that caches the information around the network currency.
    */
   private readonly networkCurrencyStorage = new SimpleObjectStorage<NetworkCurrencyModel[]>(
     'networkCurrencyStorage')
 
+  /**
+   * Gets currency mosaicId and harvesting mosaicId from the stored network configuration
+   * @returns {MosaicId[]}
+   */
+  private getNetworkCurrenciesIds(): MosaicId[] {
+    const {networkConfiguration} = new SimpleObjectStorage<NetworkModel>('network').get()
```

Comment:
My plan is to keep to only one place who can own the database/storage. NetworkService in this case. The service should take the network configuration as a parameter. Please provide the network configuration, the store can get that using the rootGetter.

Reply:
You can see the dependencies better this way, the network store may have not been executed yet so maybe the table is empty or not up date

Reply:
OK, Ill pass `getNetworkCurrenciesIds` as an argument provided by network store getters

I'll also add a node somewhere to say network store must be initialized before mosaic store